src/dsql/ExprNodes.cpp

```diff
@@ -4735,6 +4735,98 @@ ValueExprNode* CurrentRoleNode::dsqlPass(DsqlCompilerScratch* dsqlScratch)
 //--------------------
 
 
+static RegisterNode<CurrentSchemaNode> regCurrentSchemaNode({blr_current_schema});
+
+DmlNode* CurrentSchemaNode::parse(thread_db* /*tdbb*/, MemoryPool& pool, CompilerScratch* /*csb*/,
+	const UCHAR /*blrOp*/)
+{
+	return FB_NEW_POOL(pool) CurrentSchemaNode(pool);
+}
+
+string CurrentSchemaNode::internalPrint(NodePrinter& printer) const
+{
+	ValueExprNode::internalPrint(printer);
+
+	return "CurrentSchemaNode";
+}
+
+void CurrentSchemaNode::setParameterName(dsql_par* parameter) const
+{
+	parameter->par_name = parameter->par_alias = "CURRENT_SCHEMA";
+}
+
+void CurrentSchemaNode::genBlr(DsqlCompilerScratch* dsqlScratch)
+{
+	dsqlScratch->appendUChar(blr_current_schema);
+}
+
+void CurrentSchemaNode::make(DsqlCompilerScratch* dsqlScratch, dsc* desc)
+{
+	desc->dsc_dtype = dtype_varying;
+	desc->dsc_scale = 0;
+	desc->dsc_flags = DSC_nullable;
+	desc->dsc_ttype() = ttype_metadata;
+	desc->dsc_length = MAX_SQL_IDENTIFIER_LEN + sizeof(USHORT);
+}
+
+void CurrentSchemaNode::getDesc(thread_db* /*tdbb*/, CompilerScratch* /*csb*/, dsc* desc)
+{
+	desc->dsc_dtype = dtype_text;
+	desc->dsc_ttype() = ttype_metadata;
+	desc->dsc_length = MAX_SQL_IDENTIFIER_LEN;
+	desc->dsc_scale = 0;
+	desc->dsc_flags = DSC_nullable;
```

Comment:
I'm not sure I understand this `CHAR` vs `VARCHAR` difference in `jrd` vs `dsql`, but I see the same code for `CURRENT_USER` too, so OK. But if you can explain, please do.

Also, I'd suggest using `makeText() / makeVarying() `for simplicity.